relay-cabi/src/core.rs

```diff
@@ -135,7 +135,7 @@ impl From<Uuid> for RelayUuid {
 #[unsafe(no_mangle)]
 #[relay_ffi::catch_unwind]
 pub unsafe extern "C" fn relay_uuid_is_nil(uuid: *const RelayUuid) -> bool {
-    if let Ok(uuid) = Uuid::from_slice(unsafe { &(*uuid).data[..] }) {
+    if let Ok(uuid) = Uuid::from_slice(unsafe { &(&(*uuid).data)[..] }) {
```

Comment:
This looks like the wrong fix, as the lint suggets:

```
       --> relay-cabi/src/core.rs:138:50
        |
    138 |     if let Ok(uuid) = Uuid::from_slice(unsafe { &(*uuid).data[..] }) {
        |                                                  ^^^^^^^^^^^^^^^^
        |
        = note: creating a reference requires the pointer target to be valid and imposes aliasing requirements
        = note: `#[warn(dangerous_implicit_autorefs)]` on by default
    help: try using a raw pointer method instead; or if this reference is intentional, make it explicit
        |
    138 |     if let Ok(uuid) = Uuid::from_slice(unsafe { &(&(*uuid).data)[..] }) {
        |                                                  ++            +
```

Can we not avoid the reference?

Reply:
Oh nice, was trying to use u128 but that turns out does not have a [stable ABI](https://github.com/rust-lang/lang-team/issues/255) (today I learned)